File: frontend/src/Service/CourseService.js
import axios from 'axios';

const COURSE_BASE_URL = 'http://localhost:8080/springToolSuite/api/v4/Courses';

class CourseService {
    addCourse(course){
        return axios.post(COURSE_BASE_URL,course);
    }

    updateCourse(id,course){
        return axios.put(COURSE_BASE_URL+'/'+id,course)
    }
    getCourses() {
        return axios.get(COURSE_BASE_URL);
    }

    getCourseById(id) {
        return axios.get(COURSE_BASE_URL + '/' + id);
    }

    deleteCourse(id){
        return axios.delete(COURSE_BASE_URL+'/'+id)
    }
}

const courseService = new CourseService();

export default courseService;

File: Backend/src/main/java/com/stdtc/Course/CourseService.java
package com.stdtc.Course;

import java.util.List;

public interface CourseService {

	public Course save(Course course);

	public List<Course> getAll();

	public Course getById(int id);

	public boolean deleteCourse(int id);
	
	public Course updateCourse(int id,Course course);

}
